File: 2169-count-operations-to-obtain-zero/2169-count-operations-to-obtain-zero.cs
public class Solution {
    public int CountOperations(int num1, int num2) {
        var ans = 0;

        while(num1 != 0 && num2 != 0) {
            ans += num1 / num2; //  we may have to subtract num2 multiple times, so taking shortcut
            num1 %= num2;
            (num1, num2) = (num2, num1);    // swap numbers
        }

        return ans;
    }
}

File: 2169-count-operations-to-obtain-zero/2169-count-operations-to-obtain-zero.py
class Solution:
    def countOperations(self, num1: int, num2: int) -> int:
        ans = 0

        while num1 and num2:
            ans += num1 // num2 #  we may have to subtract num2 multiple times, so taking shortcut
            num1 %= num2
            num1, num2 = num2, num1    # swap numbers

        return ans